modules/clightning-db-replication.nix

```diff
@@ -0,0 +1,223 @@
+          `$secretsDir/clightning-replication-ssh`.
+          The SSH server must allow logins via this key.
+          I.e., the `authorized_keys` file of `user` must contain
+          `$secretsDir/clightning-replication-ssh.pub`.
+        '';
+      };
+      port = mkOption {
+        type = types.port;
+        default = 22;
+        description = "SSH port of the remote server.";
+      };
+      sshOptions = mkOption {
+        type = with types; listOf str;
+        default = [ "reconnect" "ServerAliveInterval=50" ];
+        description = "SSH options used for mounting the SSHFS.";
+      };
+    };
+    local = {
+      directory = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/backup/clightning";
+        description = ''
+          This option can be specified instead of `sshfs.destination` to enable
+          replication to a local directory.
+
+          If `local.setupDirectory` is disabled, the directory
+            - must already exist when `clightning.service` (or `clightning-replication-mounts.service`
+              if `encrypt` is `true`) starts.
+            - must have write permissions for the `clightning` user.
+
+          This option is also useful if you want to use a custom remote destination,
+          like a NFS or SMB share.
+        '';
+      };
+      setupDirectory = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Create `local.directory` if it doesn't exist and set write permissions
+          for the `clightning` user.
+        '';
+      };
+    };
+    encrypt = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to encrypt the replicated database with gocryptfs.
+        The encryption password is automatically generated and stored
+        in file `$secretsDir/clightning-replication-password`.
+      '';
+    };
+  };
+
+  cfg = config.services.clightning.replication;
+  inherit (config.services) clightning;
+
+  secretsDir = config.nix-bitcoin.secretsDir;
+  network = config.services.bitcoind.makeNetworkName "bitcoin" "regtest";
+  user = clightning.user;
+  group = clightning.group;
+
+  useSshfs = cfg.sshfs.destination != null;
+  useMounts = useSshfs || cfg.encrypt;
+
+  localDir = cfg.local.directory;
+  mountsDir = "/var/lib/clightning-replication";
+  sshfsDir = "${mountsDir}/sshfs";
+  plaintextDir = "${mountsDir}/plaintext";
+  destDir =
+    if cfg.encrypt then
+      plaintextDir
+    else if useSshfs then
+      sshfsDir
+    else
+      localDir;
+in {
+  inherit options;
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = useSshfs || (localDir != null);
+        message = ''
+          services.clightning.replication: One of `sshfs.destination` or
+          `local.directory` must be set.
+        '';
+      }
+      { assertion = !useSshfs || (localDir == null);
```

Comment:
I think this would actually make things more complex, because not only do you need to state and sshfs destination and port, but also directory and ssh options. For local directory it is also relevant if you want to have the service set up the directory automatically or if you want to set it up manually. The two are separate methods, and I think we should keep them as such.

Reply:
The correct type would be `oneOf submodule`. Compared to using assertions this is more complex, less readable and unexportable by `optionAttrSetToDocList` which is used when generating option docs.

Reply:
Okay, sad.